plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/azure/AzureBlobFileSystemExchangeStorage.java

```diff
@@ -99,12 +99,20 @@ public AzureBlobFileSystemExchangeStorage(ExchangeAzureConfig config)
         BlobServiceClientBuilder blobServiceClientBuilder = new BlobServiceClientBuilder()
                 .retryOptions(new RequestRetryOptions(RetryPolicyType.EXPONENTIAL, config.getMaxErrorRetries(), (Integer) null, null, null, null));
         Optional<String> connectionString = config.getAzureStorageConnectionString();
+        Optional<String> endpoint = config.getAzureStorageEndpoint();
+
+        if ((connectionString.isEmpty() && endpoint.isEmpty()) || (connectionString.isPresent() && endpoint.isPresent())) {
+            throw new IllegalArgumentException("Exactly one of exchange.azure.endpoint or exchange.azure.connection-string must be provided");
+        }
+
         if (connectionString.isPresent()) {
             blobServiceClientBuilder.connectionString(connectionString.get());
         }
         else {
+            blobServiceClientBuilder.endpoint(endpoint.get());
```

Comment:
With this change we now HAVE TO set endpoint. Is it really obligatory? Should we allow 3 options:
* connectionString set
* endpoint set
* nothing set
???

Full disclosure: I am not super familiart with Azure cloud.

Reply:
In the `BlobServiceClientBuilder` class there is a [variable](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java#L76) called `endpoint`. That can only be set via either the [`endpoint()` method](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java#L171), but also is parsed as part of the [`connectionString()` method](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClientBuilder.java#L324)

That variable ends up being passed through to the async client and it asserts there must be a valid URI or else building the client fails. This is what happens today if you pass in an empty connection string (which is technically allowable today) 

So my perspective is that one of `connectionString` or `endpoint` is required for it to work, otherwise you'd hit the error you can hit today by not supplying a `connectionString`

Reply:
makes sense